src/eko/io/struct.py

```diff
@@ -338,6 +342,52 @@ class AccessConfigs:
     """The path to the permanent object."""
     readonly: bool
     "Read-only flag"
+    open: bool
+    "EKO status"
+
+    def assert_open(self):
+        """Assert operator is open.
+
+        Raises
+        ------
+        exceptions.ClosedOperator
+            if operator is closed
+
+        """
+        if not self.open:
+            raise exceptions.ClosedOperator
+
+    def assert_writeable(self, msg: Optional[str] = None):
+        """Assert operator is writeable.
+
+        Raises
+        ------
+        exceptions.ClosedOperator
+            see :meth:`assert_open`
+        exceptions.ReadOnlyOperator
+            if operators has been declared read-only
+
+        """
+        if msg is None:
+            msg = ""
+
+        self.assert_open()
+        if self.readonly:
+            raise exceptions.ReadOnlyOperator(msg)
```

Comment:
why do you need a custom message here?

Reply:
Sometimes you might want to provide a specific reason why it is read-only, if the opening happened inside EKO (there is a single use case somewhere, you can check it).